tools-local/tasks/mobile.tasks/WasmBundleTask/WasmBundleTask.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Net;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+public class WasmBundleTask : Task
+{
+    public string? InputDirectory { get; set; }
+    public string? FileName { get; set; } 
+
+    private (byte[] json_bytes, MemoryStream stream) EnumerateData() {
+        var indices = new List<object[]>();
+        var stream = new MemoryStream();
+        
+        var directoryInfo = new DirectoryInfo(InputDirectory!);
+        
+        foreach (var entry in directoryInfo.EnumerateFiles("*", SearchOption.AllDirectories)) {
+
+            var relativePath = entry.FullName.Substring(InputDirectory!.Length).Trim('/');
+            indices.Add(new object[] { relativePath, entry.Length });
+
+            using (var readStream = entry.OpenRead())
+                readStream.CopyTo(stream);
+        }
+        
+        stream.Position = 0;
+        var jsonBytes = JsonSerializer.SerializeToUtf8Bytes(indices);
+        
+        return (jsonBytes, stream);
+    }
+
+    public override bool Execute()
+    {
+        (byte[] json_bytes, MemoryStream stream) data;
+
+        if (InputDirectory == null) {
+            throw new ArgumentException("Input directory doesn't exist.");
```

Comment:
Also, instead of throwing exceptions for invalid input, use:

```csharp
Log.LogError($"Input directory '{InputDirectory}' does not exist");
return false;
```

Reply:
Since, this is a Required input now, you don't need to check for null. Instead check for `Directory.Exists(InputDirectory)`, and `LogError`, if it doesn't.